### PR TITLE
feat: Allow for non-standard provisioning URI params, eg. image/icon

### PR DESCRIPTION
**Pull request by mdp**

### CODE REVIEW

**Comment by TSMMark** [comment]
Looks great, nice! Can we expect a gem release?

**Comment by mdp** [comment]
Yeah, so the main branch has moved along a bit and this MR is not up to date. I'll try and get it out this week after bringing it back up to date.

**Comment by TSMMark** [comment]
Awesome, looking forward to it!

**Comment by mdp** [comment]
@TSMMark Can you give this a quick review and I'll publish it ASAP. Sorry for the delay on this. Thanks

File: spec/lib/rotp/hotp_spec.rb [hunk 1]
```diff
@@ -108,14 +110,29 @@
   end
 
   describe '#provisioning_uri' do
-    it 'accepts the account name' do
+    let(:hotp)    { ROTP::HOTP.new('a' * 32, name: "m@mdp.im", issuer: "Example.com") }
```
Comment (TSMMark) [inline spec/lib/rotp/hotp_spec.rb:113]: intentional whitespace? 

```suggestion
    let(:hotp) { ROTP::HOTP.new('a' * 32, name: "m@mdp.im", issuer: "Example.com") }
```

**Review by TSMMark — APPROVED** [review]
LGTM. I might be able to test this in the next few days.

**Comment by TSMMark** [comment]
During local testing I couldn't get any image to show up my the authenticator app, even though the image param was present in the `otpauth://` url... 

Not sure if there's some specification for the image for it to show up in an authenticator app. I'm using this OTP Auth app: https://apps.apple.com/us/app/otp-auth/id659877384

**Comment by TSMMark** [comment]
i'm not very familiar with the protocol, is this the proper encoding of the url attributes?

Screen Shot 2022-08-17 at 12 11 16 PM

**Comment by mdp** [comment]
That last one definitely looks odd. Here's what I'm getting from the rotp with the image param:

``` ruby
ROTP::TOTP.new(TEST_SECRET, name: "m@mdp.im", issuer: "Example.com",
          provisioning_params: { image: 'https://assets.stickpng.com/thumbs/580b57fcd9996e24bc43c53e.png' }
)
```

`otpauth://totp/Example.com:m%40mdp.im?secret=JBSWY3DPEHPK3PXP&issuer=Example.com&image=https%3A%2F%2Fassets.stickpng.com%2Fthumbs%2F580b57fcd9996e24bc43c53e.png`

Which I encoded to:
image

And here's what I get in FreeOTP:

3f19fb0d-ae8d-48ee-be81-58ca9356d340

From what I can tell, the image doesn't get used in Authy, and Google Authenticator doesn't do logo/images at all.

**Comment by TSMMark** [comment]
I'm using OTP Auth app https://apps.apple.com/us/app/otp-auth/id659877384 and many of the accounts have images / logos. 

But maybe it's possible that there are some "presets" in the app, where they have some static images pre-defined based on well-known providers like 1password, binance, discord, github, google, amazon, etc

**Comment by mdp** [comment]
Yeah, that would be my guess. The only references I can find to the image param are for FreeOTP, which I got to work with the image param.

https://github.com/pyauth/pyotp/issues/96
https://stackoverflow.com/questions/41232168/token-image-in-google-authenticator-or-freeotp
https://stackoverflow.com/questions/62303077/mfa-one-time-password-image

**Pull request merged**